src/Resources/TransactionTypes/AbstractTransactionType.php

```diff
@@ -176,4 +176,49 @@ public function fetchPayment(): void
             $this->fetchResource($payment);
         }
     }
+
+    /**
+     * @param stdClass $response
+     *
+     * @return void
+     */
+    protected function handleAdditionalTransactionData(stdClass $response): void
+    {
+        if (isset($response->additionalTransactionData)) {
+            $this->setAdditionalTransactionData($response->additionalTransactionData);
+
+            $this->handleRiskData($response);
+            $this->handleShipping($response);
+        }
```

Comment:
It is possible to save time here because the same thing is happening several times.

```suggestion
        if (isset($response->additionalTransactionData)) {
            $additionalTransactionData = $response->additionalTransactionData;
            
            $this->setAdditionalTransactionData($additionalTransactionData);
            $this->handleRiskData($additionalTransactionData);
            $this->handleShipping($additionalTransactionData);
        }
```